### DISCUSSION TITLE
filter plugin if input source lost connection then stream stop

### DISCUSSION THREAD

**Discussion by nocpbxchap** [ant-media/Ant-Media-Server#6368] [Plugins/Custom Extensions]
hello, i was testing filter plugin the issue i found is if one of input sources lost connection then stream stop, there is a way to try to reconnect not stream stop.

example

curl -i -X POST -H "Accept: application/json" -H "Content-Type: application/json" "http://localhost:5080/WebRTCAppEE/rest/v2/filters/create" -d '{
  "filterId": "filter",
  "inputStreams": ["DiMNsHiFZxbXXzra", "qutIjZJBjxHmxcDB", "wfrTCiwIcKBmJAPW"],
  "outputStreams": ["1709760915"],
  "videoFilter": "[in0]scale=354:234:force_original_aspect_ratio=decrease,pad=360:240:3:3:color=black[s0];[in1]scale=354:234:force_original_aspect_ratio=decrease,pad=360:240:3:3:color=black[s1];[in2]scale=174:114:force_original_aspect_ratio=decrease,pad=720:120:273:3:color=black[s2];[s0][s1]hstack=inputs=2[p];[p][s2]vstack=inputs=2[out0]",
 "audioFilter": "[in0][in1][in2]amix=inputs=3[out0]",
  "videoEnabled": true,
  "audioEnabled": true,
  "type": "asynchronous"
}'

if  "inputStreams": ["DiMNsHiFZxbXXzra disconnect then stream stop.

**Comment by amarantmeida** [comment]
@nocpbxchap To achieve this, we would need to make a small development, we will convert this into an issue. How critical is this for you?

**Comment by nocpbxchap** [comment]
hi, we are triyng to make new development using filter , but we find these issue maybe if you add

1.- When creating Filter Enter Backup StreamID rule, IF ANy of Sources are Lost then Play stream XXXXXX.

And yes is important i think we can do lot of things with filter.

**Comment by amarantmeida** [comment]
@nocpbxchap We have created an issue for this and our team will look into it. You can follow it here- https://github.com/ant-media/Ant-Media-Server/issues/6372

**Comment by nocpbxchap** [comment]
Thanks, we hope it can be done quick, filter plugin has a lot of advantages
and we can add more functionality to streams.

**Comment by burak-58** [comment]
Hi @nocpbxchap,
I want to suggest a workaround solution for you. You may get stream stop status via webhook and change the filter text according to the new status.
